x-pack/plugin/esql/arrow/shim/src/main/java/org/elasticsearch/xpack/esql/arrow/shim/Shim.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.arrow.shim;
+
+import org.apache.arrow.memory.AllocationManager;
+import org.apache.arrow.memory.ArrowBuf;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.DefaultAllocationManagerOption;
+import org.elasticsearch.logging.LogManager;
+
+import java.lang.reflect.Field;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+/**
+ * We don't actually <strong>use</strong> Arrow's memory manager, but , arrow
+ * won't initialize properly unless we configure one. We configure an "empty"
+ * one here.
+ */
+public class Shim implements AllocationManager.Factory {
+    /**
+     * Initialize the Arrow shim. Arrow does some interesting reflection stuff on
+     * initialization. We can avoid it if we
+     */
+    public static void init() {
```

Comment:
Arrow has some "interesting" code in the initialization. In an effort to be magic it looks around the in classpath and calls `setAccessible` on something. That something is *public* already, but the security manager still fails.

We can avoid the whole classpath scanning and reflection magic with some contained magic of our own - this stuff. But we need security manager privileges. So I moved this to it's own tiny jar.

Lastly, this replaces all the things arrow does with the Unsafe with a shim that does nothing. That's fine for how we're using it.